lib/identity_cache/memoized_cache_proxy.rb

```diff
@@ -30,34 +30,47 @@ def with_memoization(&block)
     end
 
     def write(key, value)
-      memoized_key_values[key] = value if memoizing?
-      @cache_fetcher.write(key, value)
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.write', memoizing: memoizing) do
+        memoized_key_values[key] = value if memoizing
+        @cache_fetcher.write(key, value)
+      end
     end
 
     def delete(key)
-      memoized_key_values.delete(key) if memoizing?
-      result = @cache_fetcher.delete(key)
-      IdentityCache.logger.debug { "[IdentityCache] delete #{ result ? 'recorded'  : 'failed'  } for #{key}" }
-      result
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.delete', memoizing: memoizing) do
+        memoized_key_values.delete(key) if memoizing
+        result = @cache_fetcher.delete(key)
+        IdentityCache.logger.debug {"[IdentityCache] delete #{ result ? 'recorded' : 'failed'  } for #{key}"}
+        result
+      end
     end
 
     def fetch(key)
       used_cache_backend = true
       missed = false
-      value = if memoizing?
-        used_cache_backend = false
-        memoized_key_values.fetch(key) do
-          used_cache_backend = true
-          memoized_key_values[key] = @cache_fetcher.fetch(key) do
+      memoizing = memoizing?
+      value = ActiveSupport::Notifications.instrument('identity_cache.backend.fetch', { memoizing: memoizing }) do |payload|
```

Comment:
Right, I get that I am... Currently these `#fetch(_multi)` measure (up to) 3 things:
 1. memoized look up (if memoizing, whether hit or not, that's a HashTable lookup); then, in case of miss
 1. memcached look up (again, that's a network hit, plus the actual look up); then, in case of miss again
 1. we hit the SoR... which again could "miss".

Now the issue being that, given the "cache-through pattern", we may or may not hit either tier (threadlocal, or memcached) and even maybe even the database in a single of these metrics; I see we have a couple of choices: 

 1. Not measure at the "outer most" layer and measure each of these 3 separately (though I'm not sure the SoR time should be measured here in that case; so I'm arguing for thread-local and/or memcached); or
 1. Measure the outer layer, but measure time spent in each of the tiers and publish all of that information in the `payload` (i.e. what's there, plus the actual times); or
 1. Measure the outer layer, plus each individual layers ("nested stats"); or
 1. What I have there, only the overall time spent, plus a break down.
 1. Any other option?

I wasn't happy about option 3 (the current one), so that's the discussion I wanted to trigger... but wasn't probably very clear. wdyt?